vault/policy.go

```diff
@@ -84,6 +85,40 @@ type Permissions struct {
 	DeniedParameters   map[string][]interface{}
 }
 
+func (p *Permissions) Clone() (*Permissions, error) {
+	ret := &Permissions{
+		CapabilitiesBitmap: p.CapabilitiesBitmap,
+		MinWrappingTTL:     p.MinWrappingTTL,
+		MaxWrappingTTL:     p.MaxWrappingTTL,
+	}
+
+	switch {
+	case p.AllowedParameters == nil:
```

Comment:
Jeff why do we need both a nil check and a len == 0 check, Can you please add a comment to help understand it?